share/share.go

```diff
@@ -38,13 +29,13 @@ var (
 // on it.
 type Share = []byte
 
-// ID gets the namespace ID from the share.
-func ID(s Share) namespace.ID {
+// GetNamespace slices Namespace out of the Share.
+func GetNamespace(s Share) Namespace {
 	return s[:NamespaceSize]
 }
 
-// Data gets data from the share.
-func Data(s Share) []byte {
+// GetData slices out data of the Share.
+func GetData(s Share) []byte {
 	return s[NamespaceSize:]
 }
```

Comment:
Definitely, and I tried that a long ago, but then there is type conversion that we have to do everywhere from []Share to [][]byte. Unfortunately, Go does not do that automatically, and there are too many conversions we have to make. When it's an alias, it's converted automatically. I am still open and want them to be methods and we should open an issue. But the only way to fix conversions is to make NMT and RSMT2D use the Share type, which is controversial.

Reply:
I think it is worth an issue to make it even cleaner. [][]byte is always confusing because there only naming that shows what it is stored inside, we can utilise type system for that too